src/qt/appearancewidget.cpp

```diff
@@ -158,11 +158,13 @@ void AppearanceWidget::updateWeightSlider()
     ui->fontWeightBoldSlider->setMinimum(0);
     ui->fontWeightBoldSlider->setMaximum(nMaximum);
 
-    if (nMaximum < 4) {
-        updateFontWeightNormal(0, true);
-        updateFontWeightBold(nMaximum, true);
-    } else {
-        updateFontWeightNormal(1, true);
-        updateFontWeightBold(4, true);
+    if (fForce || !GUIUtil::isSupportedWeight(prevWeightNormal) || !GUIUtil::isSupportedWeight(prevWeightBold)) {
+        QFont::Weight weightNormal = GUIUtil::getSupportedFontWeightNormalDefault();
+        QFont::Weight weightBold = GUIUtil::getSupportedFontWeightBoldDefault();
+        int nIndexNormal, nIndexBold;
+        assert(GUIUtil::supportedWeightToIndex(weightNormal, nIndexNormal));
+        assert(GUIUtil::supportedWeightToIndex(weightBold, nIndexBold));
```

Comment:
You should not execute smth that should/could change the state inside of asserts.